esrally/driver/runner.py

```diff
@@ -434,7 +434,7 @@ def simple_stats(self, bulk_size, response):
         if response["errors"]:
             for idx, item in enumerate(response["items"]):
                 data = next(iter(item.values()))
-                if data["status"] > 299 or data["_shards"]["failed"] > 0:
+                if data["status"] > 299 or ('_shards' in data and data["_shards"]["failed"] > 0):
```

Comment:
Rally has an extensive set of unit tests. We'd like to have an additional test for this change, you can take a look at https://github.com/elastic/rally/blob/abf1623e528e3af7cec472762772021fab2e1e32/tests/driver/runner_test.py#L262-L365 for inspiration.